lib/Option/features.json

```diff
@@ -40,7 +40,7 @@
       "name": "ld-path-driver-option"
     },
     {
-      "name": "cache-compile-job"
+      "name": "compilation-caching"
```

Comment:
> As a result of swift scan API change, rename the feature in JSON file to avoid version confusion between swift-driver and libSwiftScan.

Should we embed a version number into the dictionary instead of changing the string? That would make it more obvious which string is the new vs. old API.

Alternatively, could you just dlsym the new API and use it when available? That's what we've typically done for libclang

Reply:
I removed the `dlsym` check in swiftDriver because it is getting bigger and bigger. Want to take this as a last chance to reset the feature before start compatibility support as I don't want to keep both code paths to support old and new API in swift driver.

Reply:
Any strong opinion going either way?

Reply:
I don't feel strongly enough about this to block the change. If this is the last time we need to make an incompatible API change it's fine.

Reply:
Yes, one last time. Thanks!